azkaban-hadoop-security-plugin/src/main/java/azkaban/hive/HiveMetaStoreClientFactory.java

```diff
@@ -0,0 +1,113 @@
+/*
+ * Copyright 2017 LinkedIn Corp.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ *
+ */
+
+package azkaban.hive;
+
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static java.util.Objects.requireNonNull;
+import static org.apache.hadoop.hive.metastore.api.hive_metastoreConstants.META_TABLE_STORAGE;
+
+import org.apache.commons.lang.StringUtils;
+import org.apache.commons.pool2.BasePooledObjectFactory;
+import org.apache.commons.pool2.PooledObject;
+import org.apache.commons.pool2.impl.DefaultPooledObject;
+import org.apache.hadoop.hive.conf.HiveConf;
+import org.apache.hadoop.hive.metastore.HiveMetaHookLoader;
+import org.apache.hadoop.hive.metastore.HiveMetaStoreClient;
+import org.apache.hadoop.hive.metastore.IMetaStoreClient;
+import org.apache.hadoop.hive.metastore.RetryingMetaStoreClient;
+import org.apache.hadoop.hive.metastore.api.MetaException;
+import org.apache.hadoop.hive.ql.metadata.HiveException;
+import org.apache.hadoop.hive.ql.metadata.HiveStorageHandler;
+import org.apache.hadoop.hive.ql.metadata.HiveUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+
+/**
+ * An implementation of {@link BasePooledObjectFactory} for {@link IMetaStoreClient}.
+ */
+public class HiveMetaStoreClientFactory extends BasePooledObjectFactory<IMetaStoreClient> {
+
+  public static final String HIVE_METASTORE_TOKEN_SIGNATURE = "hive.metastore.token.signature";
+  private static final Logger LOG = LoggerFactory.getLogger(HiveMetaStoreClientFactory.class);
+  private final HiveConf hiveConf;
+
+  public HiveMetaStoreClientFactory() {
+    this(new HiveConf());
+  }
+
+  public HiveMetaStoreClientFactory(final String hcatUri) {
+    this(getHiveConf(hcatUri));
+  }
+
+  public HiveMetaStoreClientFactory(final HiveConf hiveConf) {
+    this.hiveConf = hiveConf;
+  }
+
+  private static HiveConf getHiveConf(final String hcatURI) {
+    requireNonNull(hcatURI);
+    checkArgument(StringUtils.isNotBlank(hcatURI));
+
+    final HiveConf hiveConf = new HiveConf();
+    hiveConf.setVar(HiveConf.ConfVars.METASTOREURIS, hcatURI);
+    hiveConf.set(HIVE_METASTORE_TOKEN_SIGNATURE, hcatURI);
+
+    return hiveConf;
+  }
+
+  private IMetaStoreClient createMetaStoreClient() throws MetaException {
+    final HiveMetaHookLoader hookLoader = tbl -> {
```

Comment:
I am not familiar with this hive code. Is there a common utility function or an overload that will do this?  It doesn't look like it is specific to our use case.